File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float chg;
    int cents, remain, coins;

    int q = 0;
    int d = 0;
    int n = 0;
    int p = 0;

    do
    {
        // Prompt user for a positive float
        chg = get_float("How much change?\n");
    }
    // Only run if user input is a postive number
    while (chg < 0);

    // Turn cents into an intergar to avoid errors when calculating
    cents = round(chg * 100);

    // Get the number of quarters needed
    q += cents / 25;
    cents %= 25;

    // Get the number of dimes needed
    d += cents / 10;
    cents %= 10;

    // Get the number of nickels needed
    n += cents / 5;
    cents %= 5;

    // Get the number of pennies needed
    p += cents;

    printf("Change owed: %.2f\n", chg);

    // Add the number of each coin needed to get total coins
    coins = (q + d + n + p);

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
# Imports
from cs50 import get_float
from sys import exit

# Prompt user for input
chg = get_float('How much change is owed? ')

# If user enters invalid imput, re-prompt
while chg < 0:
    chg = get_float('How much change is owed? ')

# If user input is valid
else:
    q, d, n, p = 0, 0, 0, 0  # Assign counters for each coin
    cents = chg * 100  # Turn into integers to avoid errors
    
    q += cents // 25  # increase q the number of times 25 goes into cents
    cents %= 25  # get the remainder of cents left
    
    d += cents // 10  # increase d the number of times 10 goes into cents
    cents %= 10  # get the remainder of cents left
    
    n += cents // 5  # increase n the number of times 5 goes into cents
    cents %= 5  # get the remainder of cents left
    
    p += cents  # increase p by the number cents remaining
    
    print('Change owed: %.2f' % chg)  # Print change owed in the proper format
    
    coins = (q + d + n + p)  # add up all the counters for the coins
    
    print('%d' % coins)  # Return the least amount of coins to make change
